Library/ViewModels/ReportProjectFormViewModelTests.swift

```diff
@@ -1,43 +1,65 @@
 @testable import KsApi
 @testable import Library
-import ReactiveExtensions
-import ReactiveExtensions_TestHelpers
-import ReactiveSwift
+
 import XCTest
 
 final class ReportProjectFormViewModelTests: TestCase {
-  fileprivate let vm: ReportProjectFormViewModelType = ReportProjectFormViewModel()
-
-  private let userEmail = TestObserver<String, Never>()
-
   // MARK: Computed Properties
 
   private var userEmailSuccessMockService: MockService {
-    let fetchUserQueryData = GraphAPI.FetchUserQuery
+    let fetchUserEmailQueryData = GraphAPI.FetchUserEmailQuery
       .Data(unsafeResultMap: GraphUserEnvelopeTemplates.userJSONDict)
-    guard let envelope = UserEnvelope<GraphUser>.userEnvelope(from: fetchUserQueryData) else {
+
+    guard let envelope = UserEnvelope<GraphUserEmail>.userEnvelope(from: fetchUserEmailQueryData) else {
       return MockService()
     }
 
     let mockService = MockService(
-      fetchGraphUserResult: .success(envelope)
+      fetchGraphUserEmailResult: .success(envelope)
     )
 
     return mockService
   }
 
-  override func setUp() {
-    super.setUp()
-
-    self.vm.outputs.userEmail.observe(self.userEmail.observer)
-  }
-
   func testEmailText_AfterFetchingUsersEmail() {
+    let vm = ReportProjectFormViewModel(
+      projectID: "123", projectURL: "https://www.kickstarter.com/projects/foo/bar",
+      projectFlaggingKind: GraphAPI.FlaggingKind.commentDoxxing
+    )
+
     withEnvironment(apiService: self.userEmailSuccessMockService) {
-      self.vm.inputs.viewDidLoad()
+      let userEmail = CombineTestObserver<String?, Never>()
+      userEmail.observe(vm.$retrievedEmail)
+
+      XCTAssertEqual(userEmail.events.count, 1)
```

Comment:
Interesting implementation detail: `@Published` properties emit their initial value when a subscriber attaches. So retrieving the e-mail triggers _two_ events - the first containing `nil`, the second containing the actual e-mail.